### PR TITLE
[Enhancement] Inner join derive on columns is not null

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

like SQL:
```
select * 
from t0 left outer join t1 on t0.v3 = t1.v6
                   inner  join t2 on t1.v5 = t2.v8
```

it's equals as
```
select * 
from t0 inner join t1 on t0.v3 = t1.v6
             inner join t2 on t1.v5 = t2.v8
```

because t2 inner join t1 will required t1.v5 is not null, like to
```
select * 
from (
    select * 
    from t0 left outer join t1 on t0.v3 = t1.v6
    where t1.v5 is not null) inner join t2 on t1.v5 = t2.v8
```

### CODE REVIEW

**Comment by kangkaisen** [comment]
What's the performance test result for this PR?

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 63 / 65 (96.92%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownJoinPredicateBase.java|32|34|94.12%|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalJoinOperator.java|8|8|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateJoinRule.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/Utils.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/ConstantOperator.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/scalar/SimplifiedScanColumnRule.java|5|5|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateScanRule.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownJoinOnClauseRule.java|4|4|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarEquivalenceExtractor.java|7|7|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateScanRule.java [hunk 1]
```diff
@@ -52,6 +52,7 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
         Preconditions.checkState(predicates != null);
         predicates = scalarOperatorRewriter.rewrite(predicates,
                 ScalarOperatorRewriter.DEFAULT_REWRITE_SCAN_PREDICATE_RULES);
+        predicates = Utils.transTrue2Null(predicates);
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateScanRule.java:55]: Why need true to null?

Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateScanRule.java:55]: predicate is true means don't need predicate

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 65 / 65 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalJoinOperator.java|8|8|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateJoinRule.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownJoinPredicateBase.java|34|34|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/Utils.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/ConstantOperator.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/scalar/SimplifiedScanColumnRule.java|5|5|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateScanRule.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownJoinOnClauseRule.java|4|4|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/ScalarEquivalenceExtractor.java|7|7|100.00%|

**Review by kangkaisen — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**